src/app/client/src/app/modules/dashboard/components/course-dashboard/course-dashboard.component.ts

```diff
@@ -1,28 +1,120 @@
+  courseId: string;
+  queryParams: {};
+  telemetryImpression: IImpressionEventInput;
+  state: {};
 
-  constructor(private courseProgressService: CourseProgressService) { }
+  constructor(
+    private courseProgressService: CourseProgressService,
+    private activatedRoute: ActivatedRoute,
+    private userService: UserService,
+    private router: Router,
+    private navigationhelperService: NavigationHelperService,
+    public resourceService: ResourceService,
+    private toasterService: ToasterService,
+    ) { }
 
   ngOnInit() {
-  this.getBatchList();
+    this.state = history.state;
+    this.initializeFields();
+    this.courseId = _.get(this.activatedRoute.snapshot.params, 'courseId') || _.get(this.state, 'id');
+    this.queryParams = this.activatedRoute.snapshot.queryParams;
+    this.getBatchList();
+    this.setImpressionEvent();
+  }
+
+  // To initialize the dashboard data
+  initializeFields() {
+    this.resourceService.languageSelected$.pipe(takeUntil(this.unsubscribe$))
+    .subscribe(item => {
+    this.dashBoardItems = {
+      totalBatches: {
+        title: this.resourceService.frmelmnts.lbl.totalBatches,
+        count: 0,
+        type: 'small',
+      },
+      totalEnrollment: {
+        title: this.resourceService.frmelmnts.lbl.totalEnrollments,
+        count: 0,
+        type: 'small',
+      },
+      totalCompleted: {
+        title: this.resourceService.frmelmnts.lbl.totalCompletions,
+        count: 0,
+        type: 'large'
+      }
+    };
+    });
   }
 
+
+  // To set telemetry impression
+  setImpressionEvent() {
+    this.telemetryImpression = {
+      context: {
+        env: _.get(this.activatedRoute.snapshot, 'data.telemetry.env')
+      },
+      edata: {
+        type: _.get(this.activatedRoute.snapshot, 'data.telemetry.type'),
+        pageid: _.get(this.activatedRoute.snapshot, 'data.telemetry.pageid'),
+        uri: this.router.url,
+        duration: this.navigationhelperService.getPageLoadTime()
+      },
+      object: {
+          id: this.courseId,
+          type: _.get(this.state, 'type'),
+          ver: `${_.get(this.state, 'ver')}` || '1.0',
+      }
+    };
+  }
+
+  // To get the loggedIn user created batch list for the course
   getBatchList() {
+    const searchParams = {
+      courseId: this.courseId,
+      status: ['0', '1', '2'],
+      createdBy: this.userService.userid
+    };
+    this.courseProgressService.getBatches(searchParams).pipe(takeUntil(this.unsubscribe$)).subscribe(data => {
+      this.getEnrollmentAndCompletedCount(_.get(data, 'result.response'));
+    }, err => {
+      this.toasterService.error(this.resourceService.messages.emsg.m0005);
+    });
+
   }
 
-  getEnrollmentAndCompletedCount() {
+  // To get the user enrolled/completed  count for the Course
+  getEnrollmentAndCompletedCount(batchList) {
+    const batches = _.get(batchList, 'content');
+    if (batches) {
+      this.dashBoardItems.totalBatches.count = _.get(batchList, 'count');
```

Comment:
Here you create IDashboard instance for each count & push it to this.dashboardItems.